### PR TITLE
Make dcnm api return data consistent

### PR DESCRIPTION
**Pull request by mikewiebe**
##### SUMMARY
This PR updates the data we return from DCNM to always use the same dictionary format for success and error cases.

```python
        info = {}
        info['RETURN_CODE'] = rc
        info['METHOD'] = method
        info['REQUEST_PATH'] = path
        info['MESSAGE'] = msg
        info['DATA'] = json_respond_data
```

Additional changes included in this PR:
* Updates to `lib/ansible/module_utils/network/dcnm/dcnm.py` to account for changes and a few `pep8` style/syntax updates.
* Updates to `lib/ansible/module_utils/network/dcnm/dcnm.py` to check errors using only the error code.  IMO other checks are not necessary.  I also added a raise for other errors but we can update as needed.

Here are some examples of data that gets returned for various rest api calls.

#### Successful POST

```
{
    'RETURN_CODE': 200,
    'METHOD': 'POST',
    'REQUEST_PATH': 'https://10.122.197.6:443/rest/interface',
    'MESSAGE': 'OK',
    'DATA': {}
}
```

#### Successful GET

```
{
    'RETURN_CODE': 200,
    'METHOD': 'GET',
    'REQUEST_PATH': 'https://10.122.197.6:443/rest/control/fabrics/msd/fabric-associations',
    'MESSAGE': 'OK',
    'DATA': [
        {
            'fabricId': 6,
            'fabricName': 'vxlan-shrishail-green-field',
            'fabricType': 'Switch_Fabric',
            'fabricState': 'standalone',
            'fabricParent': 'None',
            'fabricTechnology': 'VXLANFabric'
        }
    ]
}
```

#### ERROR CODE 500
```
[
    {
        'RETURN_CODE': 500,
        'METHOD': 'POST',
        'REQUEST_PATH': 'https://10.122.197.6:443/rest/interface',
        'MESSAGE': 'Internal Server Error',
        'DATA': [
            {
                'reportItemType': 'ERROR',
                'message': 'Invalid Serial number',
                'entity': 'HOX1821H035~loopback55',
                'line': 0,
                'column': 0
            }
        ]
    }
]
```

### CODE REVIEW

**Comment by chrisvanheuveln** [comment]
👍

**Comment by nkshrishail** [comment]
The success responses are coming back as {} while the error responses are coming as [{}], any reason why the latter is enclosed in a list?

**Comment by mikewiebe** [comment]
> The success responses are coming back as {} while the error responses are coming as [{}], any reason why the latter is enclosed in a list?

That is a good question.  Let me investigate a bit closer.

File: lib/ansible/plugins/httpapi/dcnm.py [hunk 1]
```diff
@@ -61,28 +61,33 @@ def send_request(self, method, path, json=None):
...
-            return self._response_to_json(response_value)
+                raise Exception(self._return_info(None, method, path, msg))
+            response, rdata = self.connection.send(path, json, method=method,
+                                                   headers=self.headers,
+                                                   force_basic_auth=True)
+            return self._verify_response(response, method, path, rdata)
         except Exception as e:
             eargs = e.args[0]
             if isinstance(eargs, dict) and eargs.get('METHOD'):
                 return [eargs]
```
Comment (nkshrishail) [inline lib/ansible/plugins/httpapi/dcnm.py]: This is where the failures are getting enclosed in list.

Comment (mikewiebe) [reply lib/ansible/plugins/httpapi/dcnm.py]: Good catch.  I made the change.  Here is a failure message as reported by Ansible with this change.

```console
   "msg": {
        "DATA": [
            {
                "column": 0,
                "entity": "n9k-109:Loopback90",
                "line": 0,
                "message": "SET of IP/Subnet resource is failed for Input: Fabric=wiebe55, Pool=LOOPBACK0_IP_POOL, Entity Type=DeviceInterface, Entity=FOX1821H035~loopback90, Resource=10.10.10.89. Reason: 10.10.10.89 is allocated to Switch= [FOX1821H035/n9k-109]",
                "reportItemType": "ERROR"
            }
        ],
        "MESSAGE": "Internal Server Error",
        "METHOD": "POST",
        "REQUEST_PATH": "https://10.122.197.6:443/rest/interface",
        "RETURN_CODE": 500
    }
```

Comment (mikewiebe) [reply lib/ansible/plugins/httpapi/dcnm.py]: Good catch.  I made the change.  Here is the data we get back on failure now.

```python
{
    'DATA': [{'column': 0,
           'entity': 'n9k-109:Loopback90',
           'line': 0,
           'message': 'Loopback90 is already in use.  You can modify the '
                      'config by editing the interface.',
           'reportItemType': 'ERROR'}],
     'MESSAGE': 'Internal Server Error',
     'METHOD': 'POST',
     'REQUEST_PATH': 'https://10.122.197.6:443/rest/interface',
     'RETURN_CODE': 500
}
```

**Comment by nkshrishail** [comment]
:+1:

**Pull request merged**